python/dgl/readout.py

```diff
@@ -88,65 +53,73 @@ def sum_nodes(graph, feat, weight=None):
     Create two :class:`~dgl.DGLGraph` objects and initialize their
     node features.
 
-    >>> g1 = dgl.DGLGraph()                           # Graph 1
-    >>> g1.add_nodes(2)
-    >>> g1.ndata['h'] = th.tensor([[1.], [2.]])
-    >>> g1.ndata['w'] = th.tensor([[3.], [6.]])
+    >>> g1 = dgl.graph(([0, 1], [1, 0]))              # Graph 1
+    >>> g1.ndata['h'] = th.tensor([1., 2.])
+    >>> g2 = dgl.graph(([0, 1], [1, 2]))              # Graph 2
+    >>> g2.ndata['h'] = th.tensor([1., 2., 3.])
+    
+    Sum over one graph:
 
-    >>> g2 = dgl.DGLGraph()                           # Graph 2
-    >>> g2.add_nodes(3)
-    >>> g2.ndata['h'] = th.tensor([[1.], [2.], [3.]])
+    >>> dgl.readout_nodes(g1, 'h')
+    tensor([3.])  # 1 + 2
 
-    Sum over node attribute :attr:`h` without weighting for each graph in a
-    batched graph.
+    Sum over a batched graph:
 
-    >>> bg = dgl.batch([g1, g2], node_attrs='h')
-    >>> dgl.sum_nodes(bg, 'h')
-    tensor([[3.],   # 1 + 2
-            [6.]])  # 1 + 2 + 3
+    >>> bg = dgl.batch([g1, g2])
+    >>> dgl.readout_nodes(bg, 'h')
+    tensor([3., 6.])  # [1 + 2, 1 + 2 + 3]
 
-    Sum node attribute :attr:`h` with weight from node attribute :attr:`w`
-    for a single graph.
+    Weighted sum:
 
-    >>> dgl.sum_nodes(g1, 'h', 'w')
-    tensor([[15.]]) # 1 * 3 + 2 * 6
+    >>> bg.ndata['w'] = th.tensor([.1, .2, .1, .5, .2])
+    >>> dgl.readout_nodes(bg, 'h', 'w')
+    tensor([.5, 1.7])
+    
+    Readout by max:
+    
+    >>> dgl.readout_nodes(bg, 'h', op='max')
+    tensor([2., 3.])
 
     See Also
     --------
-    mean_nodes
-    sum_edges
-    mean_edges
-    """
-    return _sum_on(graph, 'nodes', feat, weight)
+    readout_edges
+    """ 
+    x = graph.nodes[ntype].data[feat]
+    if weight is not None:
+        x = x * graph.nodes[ntype].data[weight]
+    return segment.segment_reduce(graph.batch_num_nodes(ntype), x, reducer=op)
 
-def sum_edges(graph, feat, weight=None):
-    """Sums all the values of edge field :attr:`feat` in :attr:`graph`,
-    optionally multiplies the field by a scalar edge field :attr:`weight`.
+def readout_edges(graph, feat, weight=None, *, op='sum', etype=None):
+    """Sum the edge feature :attr:`feat` in :attr:`graph`, optionally
+    multiplies it by a edge :attr:`weight`.
+
+    The function is commonly used as a *readout* function on a batch of graphs
+    to generate graph-level representation. Thus, the result tensor shape
+    depends on the batch size of the input graph. Given a graph of batch size
+    :math:`B`, and a feature size of :math:`D`, the result shape will be
+    :math:`(B, D)`, with each row being the aggregated edge features of each
+    graph.
```

Comment:
Briefly describe the behavior in the case of multiple edge types and give an example?